plugins/GSdx/stdafx.cpp

```diff
@@ -71,16 +71,73 @@ void vmfree(void* ptr, size_t size)
 	VirtualFree(ptr, 0, MEM_RELEASE);
 }
 
+static HANDLE s_fh = NULL;
+static uint8* s_Next[8];
+
 void* fifo_alloc(size_t size, size_t repeat)
 {
-	// FIXME check linux code
-	return vmalloc(size * repeat, false);
+	ASSERT(s_fh == NULL);
+
+	if (repeat >= countof(s_Next)) {
+		fprintf(stderr, "Memory mapping overflow (%zu >= %u)\n", repeat, countof(s_Next));
+		return vmalloc(size * repeat, false); // Fallback to default vmalloc
+	}
+
+	s_fh = CreateFileMapping(INVALID_HANDLE_VALUE, nullptr, PAGE_READWRITE, 0, size, nullptr);
+	DWORD errorID = ::GetLastError();
+	if (s_fh == NULL) {
+		fprintf(stderr, "Failed to reserve memory. WIN API ERROR:%u\n", errorID);
+		return vmalloc(size * repeat, false); // Fallback to default vmalloc
+	}
+
+	int mmap_segment_failed = 0;
+	void* fifo = MapViewOfFile(s_fh, FILE_MAP_ALL_ACCESS, 0, 0, size);
+	for (size_t i = 1; i < repeat; i++) {
+		void* base = (uint8*)fifo + size * i;
+		s_Next[i] = (uint8*)MapViewOfFileEx(s_fh, FILE_MAP_ALL_ACCESS, 0, 0, size, base);
+		errorID = ::GetLastError();
+		if (s_Next[i] != base) {
+			for (i; i >= 1; i--) {
```

Comment:
`i` at the init-statement condition does nothing. I would have done
```
do {
// blah
} while (--i > 0);
```
but that's just personal preference.